src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -431,3 +530,339 @@ def can_create_clusterrolebindings(configuration):
+                       "Please run 'az connectedk8s troubleshoot' command again later to get the analysis results.")
+
+
+def format_diag_status(diag_status):
+    for diag in diag_status:
+        if diag["Status"]:
+            if "Error:" in diag["Status"]:
+                diag["Status"] = f'{colorama.Fore.RED}{diag["Status"]}{colorama.Style.RESET_ALL}'
+            else:
+                diag["Status"] = f'{colorama.Fore.GREEN}{diag["Status"]}{colorama.Style.RESET_ALL}'
+
+    return diag_status
+
+
+def collect_periscope_logs(resource_group_name, name, storage_account_name=None, sas_token=None, container_name=None, readonly_sas_token=None, kube_context=None, kube_config=None):
+    colorama.init()
+
+    if not which('kubectl'):
+        raise ValidationError('Can not find kubectl executable in PATH')
+
+    kubectl_prior = ["kubectl"]
+    if kube_config:
+        kubectl_prior.extend(["--kubeconfig", kube_config])
+    if kube_context:
+        kubectl_prior.extend(["--context", kube_context])
+
+    readonly_sas_token = readonly_sas_token.strip('?')
+
+    from knack.prompting import prompt_y_n
+
+    print()
+    print('This will deploy a daemon set to your cluster to collect logs and diagnostic information and '
+          f'save them to the storage account '
+          f'{colorama.Style.BRIGHT}{colorama.Fore.GREEN}{storage_account_name}{colorama.Style.RESET_ALL} as '
+          f'outlined in {format_hyperlink("http://aka.ms/AKSPeriscope")}.')
+    print()
+    print('If you share access to that storage account to Azure support, you consent to the terms outlined'
+          f' in {format_hyperlink("http://aka.ms/DiagConsent")}.')
+    print()
+    if not prompt_y_n('Do you confirm?', default="n"):
+        return
+
+    sas_token = sas_token.strip('?')
+    deployment_yaml = urlopen(
+        "https://raw.githubusercontent.com/Azure/aks-periscope/master/deployment/aks-periscope.yaml").read().decode()
+    deployment_yaml = deployment_yaml.replace("# <accountName, base64 encoded>",
+                                              (base64.b64encode(bytes(storage_account_name, 'ascii'))).decode('ascii'))
+    deployment_yaml = deployment_yaml.replace("# <saskey, base64 encoded>",
+                                              (base64.b64encode(bytes("?" + sas_token, 'ascii'))).decode('ascii'))
+    deployment_yaml = deployment_yaml.replace("aksrepos.azurecr.io/staging/aks-periscope:v0.4", "aksrepos.azurecr.io/staging/aks-periscope:0.5")
+    container_logs = "azure-arc"
+    kube_objects = "azure-arc/pod azure-arc/service"
+    yaml_lines = deployment_yaml.splitlines()
+    for index, line in enumerate(yaml_lines):
+        if "DIAGNOSTIC_CONTAINERLOGS_LIST" in line:
+            yaml_lines[index] = line + ' ' + container_logs
+        if "AZURE_BLOB_SAS_KEY" in line:
+            yaml_lines[index] = line + '\n' + '  AZURE_BLOB_CONTAINER_NAME: ' + base64.b64encode(bytes(container_name, 'ascii')).decode('ascii')
```

Comment:
If custom container name is not provided by user, then, will it face issue in decoding?